MacOS CI workflow occasionally fails with missing m4

I could understand it if happened every time, but it only seems to happen some times. The MacOS CI runs fail like this:

```
aclocal: warning: couldn't open directory 'm4': No such file or directory
glibtoolize:   error: One of these is required:
glibtoolize:                 gm4 gnum4 m4
glibtoolize:   error: Please install GNU M4, or 'export M4=/path/to/gnu/m4'.
autoreconf: error: glibtoolize failed with exit status: 1
Error: Process completed with exit code 1.
```

--- a/.github/workflows/run_tests_osx.yml
+++ b/.github/workflows/run_tests_osx.yml
@@ -108,7 +108,7 @@ jobs:
       ###
       - name: Install autoconf
         shell: bash -l {0}
-        run: brew install automake
+        run: brew install automake m4
 
       - name: Run autoconf
         shell: bash -l {0}

